### PR TITLE
fix: Transfer: batch payment link not working properly

### LINKED ISSUE #8957: Transfer: batch payment link not working properly
I was creating batch payments in Transfer and also I wanted to create a link. But when you add two addresses with a VARIOUS amount of DOT/USD and create a recurring link, the link is not working properly because it changes the amount so it's the SAME for both addresses. Check this video:

On production

https://github.com/kodadot/nft-gallery/assets/104380776/55b5a980-6c48-49b3-9775-e270274af387

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [x] Feature
- [ ] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

**❗️  now payment link is based on display unit tab** 

- usd -> will add usdamount query param 
- token -> will add amount query param

it was not part of the original issue but I thought I'd  be nice to have

can be removed if it's not needed 

cc @JustLuuuu @yangwao 

related #7287

- [x] Closes #8957

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

https://github.com/kodadot/nft-gallery/assets/44554284/afd1efd8-0637-4268-b842-11e35d6fc67c

### CODE REVIEW

**Comment by prury** [comment]
if you had created your branch directly on nft-gallery repo i could easily work with you to modify the tests since the transfer functionality has changed 😛

File: components/transfer/Transfer.vue [hunk 1]
```diff
@@ -316,7 +316,7 @@
 <script lang="ts" setup>
 import { ApiFactory } from '@kodadot1/sub-api'
 import { ALTERNATIVE_ENDPOINT_MAP, chainNames } from '@kodadot1/static'
-
+import { zipWith } from 'lodash'
```
Comment (preschian) [inline components/transfer/Transfer.vue:319]: usually import it from specific method will give use smaller size. `'lodash/zipWith'`

File: components/transfer/Transfer.vue [hunk 2]
```diff
@@ -846,15 +890,19 @@ const onTxError = async (dispatchError: DispatchError): Promise<void> => {
   isLoading.value = false
 }
 
-const generatePaymentLink = (addressList: string[]): string => {
+const generatePaymentLink = (addressList: TargetAddress[]): string => {
   const url = new URL(`${location.origin}${location.pathname}`)
+
   addressList.forEach((addr, i) => {
-    url.searchParams.append(`target${i == 0 ? '' : i}`, addr)
+    const suffix = i == 0 ? '' : i
```
Comment (preschian) [inline components/transfer/Transfer.vue:897]:
```suggestion
    const suffix = i === 0 ? '' : i
```

**Review by preschian — APPROVED** [review]
otherwise lgtm

File: components/transfer/Transfer.vue (hunk 2 above)
Comment (hassnian) [reply components/transfer/Transfer.vue]: fixed

File: components/transfer/Transfer.vue (hunk 1 above)
Comment (hassnian) [reply components/transfer/Transfer.vue]: got it, fixed

**Review by roiLeo — APPROVED** [review]

**Review by JustLuuuu — APPROVED** [review]
works for me 🚀 thx

**Comment by yangwao** [comment]
ahh ok works 

https://deploy-preview-8965--koda-canary.netlify.app/ahp/transfer?target=16JDgnrwPLk291bh9BunucpiYKjPHvQmgHrXXMYE5ZwxDqnY&amount=50&target1=13zk78C68DHpXK96bh5MMRk8CrxcFfDQKLyAQZriLy2tzHoJ&amount1=100

**Comment by yangwao** [comment]
Thanks!
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 7.34 USD/DOT ~ 6.812 $DOT
🧗 [13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh](https://polkadot.subscan.io/account/13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)
🔗 [0x114d0c4702cdc9f5572089cb8b5877ed3b8bfd4616647888175524bdec6e2923](https://polkadot.subscan.io/extrinsic/0x114d0c4702cdc9f5572089cb8b5877ed3b8bfd4616647888175524bdec6e2923)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**